web-client/src/components/InventoryTable.tsx

```diff
@@ -0,0 +1,124 @@
+                alert(`Error deleting sock: ${error.message}`);
+            },
+        });
+    };
+
+    const openDeleteDialog = (sockId: string) => {
+        setSelectedSockId(sockId);
+        setIsDialogOpen(true);
+    };
+
+    const handleDeleteConfirm = () => {
+        if (selectedSockId) {
+            handleDeleteClick(selectedSockId);
+        }
+        setIsDialogOpen(false);
+        setSelectedSockId(null);
+    };
+
+    const handleDeleteCancel = () => {
+        setIsDialogOpen(false);
+        setSelectedSockId(null);
+    };
+
+    const calculateAveragePrice = (variants?: Variant[]) => {
+        if (!variants || variants.length === 0) return 'N/A';
+        const total = variants.reduce((acc, variant) => acc + variant.price, 0);
+        return `$${(total / variants.length).toFixed(2)}`;
+    };
+
+    const calculateTotalQuantity = (variants?: Variant[]) => {
+        if (!variants || variants.length === 0) return 0;
+        return variants.reduce((total, variant) => total + variant.quantity, 0);
+    };
+
+    return (
+        <>
+            <table className="w-full text-left border border-gray-200">
+                <thead>
+                    <tr>
+                        <th className="p-3 border-b text-center">Image</th>
+                        <th className="p-3 border-b text-center">Product ID</th>
+                        <th className="p-3 border-b">Name</th>
+                        <th className="p-3 border-b">Price</th>
+                        <th className="p-3 border-b">Stock</th>
+                        <th className="p-3 border-b">Actions</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {socks.map((sock) => (
+                        <tr
+                            key={sock.id}
+                            className="hover:bg-gray-100 cursor-pointer"
+                            onClick={() => onRowClick(sock.id)}
+                        >
+                            <td className="p-3 border-b text-center">
+                                <img src={sock.imageUrl} className="w-16 h-16 object-cover mx-auto" />
+                            </td>
+                            <td className="p-3 border-b text-center">{sock.id}</td>
+                            <td className="p-3 border-b">{sock.name}</td>
+                            <td className="p-3 border-b">
+                                {calculateAveragePrice(sock.variants)}
+                            </td>
+                            <td className="p-3 border-b">{calculateTotalQuantity(sock.variants)}</td>
+                            <td className="p-3 border-b">
+                                <Button
+                                    onClick={(e) => {
+                                        e.stopPropagation();
+                                        openDeleteDialog(sock.id);
+                                    }}
```

Comment:
Make the button be "icon sized" like this: `size="icon"`

Reply:
Updated:
<img width="430" alt="image" src="https://github.com/user-attachments/assets/a871b821-8241-4348-8a6c-d03c343805ab">